cli/chaosli/cmd/create.go

```diff
@@ -6,29 +6,618 @@
+			}
+		} else {
+			return fmt.Errorf("expected a string response, rather than type %v", reflect.TypeOf(val).Name())
+		}
+
+		return nil
+	}
+
+	name := getInput("Please name your disruption.",
+		"This will be the name used when you want to run `kubectl describe disruption`",
+		survey.WithValidator(survey.Required),
+		survey.WithValidator(validator),
+	)
+	namespace := getInput(
+		"What namespace should your disruption be created in?",
+		"If you are targeting pods, you _must_ create the disruption in the same namespace as the targeted pods.",
+		survey.WithValidator(survey.Required),
+		survey.WithValidator(validator),
+	)
+
+	return []byte(fmt.Sprintf(`{"name": %s, "namespace": %s}`, name, namespace))
+}
+
+func getDNS() v1beta1.DNSDisruptionSpec {
+	if !confirmKind("DNS Disruption", "Overrides DNS resolution for specified hostnames with a MitM DNS attack. All other DNS requests will use the target's normal DNS resolver.") {
+		return nil
+	}
+
+	getHostRecordPair := func() v1beta1.HostRecordPair {
+		hrPair := v1beta1.HostRecordPair{}
+
+		hrPair.Hostname = getInput("Specify a hostname to target",
+			"When your target makes a DNS request for this hostname; the disruption will make sure the value you specify is returned, rather than the real record.",
+			survey.WithValidator(survey.Required),
+		)
+		hrPair.Record.Type, _ = selectInput("the type of DNS record to inject",
+			[]string{"A", "CNAME"},
+			"We only support these two types of DNS requests for now. An A record request gets back an IP for a hostname, while a CNAME request maps an alias domain name to the canonical name.")
+		helpText := "We're specifying an A record, so the value should be an IP address. You can specify multiple IP addresses, if desired. Simply delimit them with commas, no whitespace! The disruption will round-robin between the options."
+
+		if hrPair.Record.Type == "CNAME" {
+			helpText = "We're specifying a CNAME record, so the value should be a hostname to redirect to."
+		}
+
+		hrPair.Record.Value = getInput("What value would you like to inject into this DNS record?", helpText, survey.WithValidator(survey.Required))
+
+		return hrPair
+	}
+
+	fmt.Println("Let's specify a DNS record to inject!")
+
+	spec := v1beta1.DNSDisruptionSpec{}
+	spec = append(spec, getHostRecordPair())
+
+	for confirmOption("Would you like to override another DNS record?", "") {
+		spec = append(spec, getHostRecordPair())
+	}
+
+	return spec
+}
+
+func getDiskPressure() *v1beta1.DiskPressureSpec {
+	if !confirmKind("Disk Pressure", "Applies IO pressure to the target") {
+		return nil
+	}
+
+	spec := &v1beta1.DiskPressureSpec{}
+
+	spec.Path = getInput(
+		"Specify a path to apply IO pressure to, e.g., /mnt/data",
+		"Specify a specific mount point to target the disk mounted there",
+		survey.WithValidator(survey.Required),
+	)
+
+	if confirmOption("Would you like to apply read pressure?", "This applies read-based IO pressure (check the docs)") {
```

Comment:
I know we use the word "pressure" exclusively, but this makes me think that we stress the disk when we do not. I think its fine to refer to it as disk pressure, but maybe for these prints (read and write) we can make it clear that its thresholds that are being put in place.

"This applies read-based IO thresholds"
"Specify the target threshold in bytes per second"
As examples.

Reply:
This is a good time for me to re-read the disk pressure docs, as I dont know how they work.